medallion/config.py

```diff
@@ -0,0 +1,104 @@
+import collections.abc
+import json
+import pathlib
+
+import attr
+import environ
+import jsonmerge
+
+from . import backends
+
+DEFAULT_CONFFILE = "/etc/medallion.conf"
+DEFAULT_CONFDIR = "/etc/medallion.d/"
```

Comment:
I'm not super familiar with configs and their standard places, but is there a better way we can make this platform agnostic? `/etc` is the obvious choice for Unix environments,  but this does not exist by default on Windows. Is there a file location that would work for both systems?

Reply:
Ah, good point. I should have twigged on that when you mentioned running the tests on Windows. IMO we'd normally want to use a base directory like `%APPDATA%/medallion/` or something similar. I might quiz a few Windows friends to see if they think that's insane or not.

Reply:
I've looked into this briefly and came across the appdirs library which seems like it'd be good to use, however it prefers to use XDG paths for Linux which means we'd get a default site config path of `/etc/xdg/medallion/` which would be disruptive for in-place upgrades. As an alternative, I could use the current paths for Linux and use appdirs on other platforms (Windows, macOS). Any thoughts?

Reply:
I'm personally against platform-dependent implementations, though I understand sometimes they are unavoidable. Any thoughts here @clenk?

Reply:
I've just pushed a commit which uses the `appdirs` library to provide sane paths for non-Linux platforms. That should give an idea of /a/ solution to the problem. Additionally, now or in future, we could add an opt-in to use XDG paths on Linux as well (ie. use appdirs on that platform too).

Happy for you to strip that commit off if you'd prefer not to use it, and I'll also keep an eye on this thread for input from you and @clenk.

Reply:
I meant that since the default path would go from `/etc/medallion.conf` in, e.g. the medallion docker image, to `/etc/xdg/medallion/<something>`. But you're right, I forgot that without this diff applied, medallion needs a specific path to a config file specified, and the docker image actually provides that path as `/opt/taxii/medallion.conf`.

I can certainly change this to use `appdirs` on all platforms and adjust any references to the `/etc/medallion{.conf,.d/}` paths.

Reply:
This is all done in the new diff, lmk what you think

Reply:
Looks really good actually. Would rather pass off the responsibility of configuration directories to that library than us handle it ourselves.